### PR TITLE
Use new `rustpython_format` crate over `rustpython-common`

### PR DESCRIPTION
**Pull request by MichaReiser**
Use the new `rustpython_format` crate over the larger `rustpython-common` crate. This removes some dependencies (335-320 for debug builds) and we now only depend on the `RustPython-parser`  repository.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4388** <a href="https://app.graphite.dev/github/pr/charliermarsh/ruff/4388" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/charliermarsh/ruff/4388?utm_source=stack-comment).

File: Cargo.toml [hunk 1]
```diff
@@ -30,11 +30,11 @@ path-absolutize = { version = "3.0.14" }
 proc-macro2 = { version = "1.0.51" }
 quote = { version = "1.0.23" }
 regex = { version = "1.7.1" }
-ruff_text_size = { git = "https://github.com/RustPython/Parser.git", rev = "947fb53d0b41fec465db3d8e725bdb2eec1299ec" }
 rustc-hash = { version = "1.1.0" }
-rustpython-common = { git = "https://github.com/RustPython/RustPython.git", rev = "f3e4d3409253660bd4fa7f3d24d3db747e7dca61" }
-rustpython-literal = { git = "https://github.com/RustPython/Parser.git", rev = "947fb53d0b41fec465db3d8e725bdb2eec1299ec" }
-rustpython-parser = { git = "https://github.com/RustPython/Parser.git", rev = "947fb53d0b41fec465db3d8e725bdb2eec1299ec" , default-features = false}
+ruff_text_size = { git = "https://github.com/RustPython/Parser.git", rev = "a983f4383fb1ad8c1c66acb1d5b0016e59f95a49" }
```
Comment (konstin) [inline Cargo.toml:34]: nit: i think these used to be alphabetically ordered, which `cargo add` likes

**Review by konstin — APPROVED** [review]

**Comment by charliermarsh** [comment]
@MichaReiser - Can I merge this and subsequent PR, to include in today's release?

**Comment by MichaReiser** [comment]
> @MichaReiser - Can I merge this and subsequent PR, to include in today's release?

Yes... but too late...

**Pull request merged**